gcs/constraints/circuit/circuit.hh

```diff
@@ -0,0 +1,130 @@
+
+
+#ifndef GLASGOW_CONSTRAINT_SOLVER_GUARD_GCS_CIRCUIT_HH
+#define GLASGOW_CONSTRAINT_SOLVER_GUARD_GCS_CIRCUIT_HH
+
+#include <gcs/constraint.hh>
+#include <gcs/innards/proof-fwd.hh>
+#include <gcs/innards/proof.hh>
+
+#include <gcs/innards/state.hh>
+#include <gcs/variable_id.hh>
+#include <list>
+#include <map>
+#include <optional>
+#include <vector>
+
+namespace gcs
+{
+    struct ProofFlagData
```

Comment:
These should be in a sub namespace, not in `gcs::`. Possibly `gcs::circuit::innards::` to avoid possible clashes with other constraints, and to make it really clear to users they shouldn't be touching it?

Reply:
You could bring `ProofFlag` into `gcs::circuit::innards` with a `using`, I guess, although that's visible for everyone using the header, which isn't great. Maybe the namespace should be `gcs::innards::circuit` rather than `gcs::circuit::innards`?

Reply:
Ah yes gcs::innards::circuit would make sense